prefetch.bs

```diff
@@ -411,6 +410,14 @@ Update all creation sites to supply an empty string, except for any in this docu
 
 <hr>
 
+Add an additional item to [=source snapshot params=] as follows:
+:  <dfn for="source snapshot params">prefetch records</dfn>
+:: a a [=list=] of [=prefetch records=].
+
+When to [=snapshot source snapshot params=], set [=source snapshot params/prefetch records=] to <var ignore>sourceDocument</var>'s [=Document/prefetch records=].
```

Comment:
Probably make a [=list/clone=] here (https://infra.spec.whatwg.org/#list-clone) to reflect the idea we're snapshotting.

Reply:
I'm not so sure: in general creating a snapshot would be a good idea to avoid unintentional race conditions, while in this case not taking a clone might be also an option,

- As [=list/clone=] is a shallow copy, this isn't a complete frozen snapshot of the prefetch records.
- When a prefetch is still ongoing at the time of navigation start and we want to use the prefetch result after response is received after navigation start, we might need intentional interactions via (non-snapshot) prefetch records (IIUC  such case is not yet specified so though)

Reply:
Yeah, I can see your concerns. However, I think it'd most fit the spirit of source snapshot params to start with a shallow clone for now, reflecting the idea that any prefetches added after the prerender navigation starts, should not influence the prerender navigation. Similar to how, e.g., changes to the page's origin or sandboxing flags after navigation starts, should not affect the (normal, non-prerender) navigation.

I agree that it's a little strange that we're not doing a deep clone. But that's kind of the nature of prefetch records being things that are updated over time. I think that part is OK.